src/asm/validate_header.c

```diff
@@ -0,0 +1,116 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   validate_header.c                                  :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: asarandi <asarandi@student.42.us.org>      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2018/02/11 20:41:20 by asarandi          #+#    #+#             */
+/*   Updated: 2018/02/28 01:35:18 by asarandi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "asm.h"
+
+void	validate_champion_name(t_asm *a, int *i)
+{
+	int		len;
+	char	*quote1;
+	char	*str;
+
+	str = ft_strdup(a->split[*i]);
+	while ((count_matching_chars(str, '"') < 2) && (*i < a->lines - 1))
+		append_nl_string(&str, a->split[++*i]);
+	if (count_matching_chars(str, '"') > 2)
+		return (header_error(".name has too many quote marks (\")", a));
+	if (count_matching_chars(str, '"') == 2)
+	{
+		quote1 = ft_strchr(str, '"') + 1;
+		len = ft_strchr(quote1, '"') - quote1;
+		if (len < 1)
+			warning(".name is blank");
+		if (len > PROG_NAME_LENGTH)
+			return (header_error(".name length exceeds PROG_NAME_LENGTH", a));
+		ft_strncpy(a->header.prog_name, quote1, len);
+		a->have_name = 1;
+	}
+	free(str);
+	return ;
+}
+
+void	validate_champion_comment(t_asm *a, int *i)
```

Comment:
This function appears to be identical to `validate_champion_name` except for the copy destination. Consider consolidating them into a single function.

Reply:
too many variables to parametrize, besides destination there is ".name"/".comment", have_name/have_comment variable, "PROG_NAME_LENGTH"/"COMMENT_LENGTH" strings;
pro/con: could be parametrized and consolidated at the cost of verbosity of error messages

Reply:
You can make it a single boolean flag.

Reply:
I would agree with consolidating these two functions.  Perhaps instead of a boolean flag, you can pass one of the macros (`PROG_NAME_LENGTH` / `COMMENT_LENGTH`) as a parameter.